lib/render.module.ts

```diff
@@ -23,6 +23,15 @@ export class RenderModule {
       await next.prepare();
     }
 
+    const nextConfig = (next as any).nextConfig;
+    const nextServer = (next as any).server;
+
+    const basePath = nextConfig
+      ? nextConfig.basePath
+      : nextServer.nextConfig.basePath;
+
+    const config = { basePath, ...options };
```

Comment:
`basePath` from RenderModule options has higher priority (in case you might want to override the base path in `next.config.js` for internal endpoints (eg your Nest application is deployed behind a proxy)).